File: obg.Domain/Entities/Medicine.cs
using obg.Domain.Enums;
using System;
using System.ComponentModel.DataAnnotations;

namespace obg.Domain.Entities
{
    public class Medicine
    {
        [Key] public string Code { get; set; }
        public string Name { get; set; }
        public string SymtompsItTreats { get; set; }
        public PresentationMedicine Presentation { get; set; }
        public int Quantity { get; set; }
        public string Unit { get; set; }
        public double Price { get; set; }
        public int Stock { get; set; }
        public bool Prescription { get; set; }
        public bool IsActive { get; set; }

        public Medicine() { }
        public Medicine(string code, string name, string symtompsItTreats, PresentationMedicine presentation, int quantity, string unit, double price, bool prescription, bool isActive)
        {
            Code = code;
            Name = name;
            SymtompsItTreats = symtompsItTreats;
            Presentation = presentation;
            Quantity = quantity;
            Unit = unit;
            Price = price;
            Stock = 0;
            Prescription = prescription;
            IsActive = isActive;
        }

    }
}

File: ObgApp/src/app/models/medicine.ts
export enum PresentationMedicine {
    capsulas,
    comprimidos,
    solucionSoluble,
    stickPack,
    liquido
}

export class Medicine {
    code: string | null;
    name: string | null;
    symtompsItTreats: string | null;
    presentation: PresentationMedicine | null;
    quantity: number | null;
    unit: string | null;
    price: number | null;
    stock: number | null;
    prescription: boolean | null;
    isActive: boolean | null;

    constructor(code: string | null, name: string | null, symtompsItTreats: string | null, presentation: PresentationMedicine | null, quantity: number | null, unit: string | null, price: number | null, stock: number | null, prescription: boolean | null, isActive: boolean | null) {
        this.code = code;
        this.name = name;
        this.symtompsItTreats = symtompsItTreats;
        this.presentation = presentation;
        this.quantity = quantity;
        this.unit = unit;
        this.price = price;
        this.stock = stock;
        this.prescription = prescription;
        this.isActive = isActive;
    }
}